src/blog/converting-svg-to-image-png-webp-etc-in-javascript.md

```diff
@@ -0,0 +1,337 @@
+---
+path: "/blog/converting-svg-to-image-png-webp-etc-in-javascript"
+date: "2021-12-23"
+title: "Converting SVG To Image (PNG, Webp, etc.) in JavaScript"
+description: "How to convert SVG to image in the browser without dependencies."
+tags: "#javascript #webdev #typescript #tutorial"
+image: "https://cdn-images-1.medium.com/max/1600/1*zXDaYiF8SRHlI_OIGZnf8w.jpeg"
+canonical: "https://zhx6p-eqaaa-aaaai-abbrq-cai.raw.ic0.app/d/converting-svg-to-image-in-javascript"
+---
+
+![](https://cdn-images-1.medium.com/max/1600/1*zXDaYiF8SRHlI_OIGZnf8w.jpeg)
+
+*Photo by [Justin Aikin](https://unsplash.com/@justnjames?utm_source=unsplash&utm_medium=referral&utm_content=creditCopyText) on [Unsplash](https://unsplash.com/s/photos/dog?utm_source=unsplash&utm_medium=referral&utm_content=creditCopyText)*
+
+*****
+
+Last Monday I built and published [a new Web Components](https://docs.deckdeckgo.com/?path=/story/components-social-image--social-image), developed with [StencilJS](https://stenciljs.com/), to generate social images dynamically in the browser.
+
+For a given text and logo, the component renders a shadowed SVG that can be converted to image (PNG, Webp, etc.) and that without any third party dependencies.
+
+This is how it works.
+
+> Note: Code snippets are written in TypeScript.
+
+*****
+
+### SVG
+
+To create the dynamic SVG, I used a `foreignObject` to embed the text with an HTML paragraph (`<p/>`) and a graphical `image` element.
+
+*****
+
+#### ForeignObject
+
+I could have use `<text/>` elements to draw graphics instead of HTML elements but, the feature has the need to support dynamic inputs. These can be too long and might need to be truncated and displayed with three ending dots `...` .
```

Comment:
```suggestion
I could have used `<text/>` elements to draw graphics instead of HTML elements, however the feature needs to support dynamic inputs. These can be too long and might need to be truncated and displayed with three ending dots `...` .
```

Reply:
What exactly does "These" refer to in "These can be too long..."?